tests/Kiota.Builder.Tests/Writers/TypeScript/CodeFunctionWriterTests.cs

```diff
@@ -1118,4 +1130,317 @@ public async Task WritesConstructorWithEnumValue()
+        var builder = new KiotaBuilder(mockLogger.Object, new GenerationConfiguration { ClientClassName = "Primitives", Serializers = ["none"], Deserializers = ["none"] }, _httpClient);
+        await using var fs = new FileStream(tempFilePath, FileMode.Open);
+        var document = await builder.CreateOpenApiDocumentAsync(fs);
+        var node = builder.CreateUriSpace(document);
+        builder.SetApiRootUrl();
+        var codeModel = builder.CreateSourceModel(node);
+        var rootNS = codeModel.FindNamespaceByName("ApiSdk");
+        Assert.NotNull(rootNS);
+        var clientBuilder = rootNS.FindChildByName<CodeClass>("Primitives", false);
+        Assert.NotNull(clientBuilder);
+        var constructor = clientBuilder.Methods.FirstOrDefault(static x => x.IsOfKind(CodeMethodKind.ClientConstructor));
+        Assert.NotNull(constructor);
+        Assert.Empty(constructor.SerializerModules);
+        Assert.Empty(constructor.DeserializerModules);
+        await ILanguageRefiner.Refine(generationConfiguration, rootNS);
+        Assert.NotNull(rootNS);
+        var modelsNS = rootNS.FindNamespaceByName("ApiSdk.primitives");
+        Assert.NotNull(modelsNS);
+        var modelCodeFile = modelsNS.FindChildByName<CodeFile>("primitivesRequestBuilder", false);
+        Assert.NotNull(modelCodeFile);
+
+        /*
+        \/**
+        * Creates a new instance of the appropriate class based on discriminator value
+        * @returns {ValidationError_errors_value}
+        *\/
+           export function createPrimitivesFromDiscriminatorValue(parseNode: ParseNode | undefined) : Primitives | undefined {
+                if (parseNode) {
+                    parseNode.getNumberValue() || parseNode.getStringValue();
+                }
+                return undefined;
+            }
+         
+         */
+
+        // Test Factory function
+        var factoryFunction = modelCodeFile.GetChildElements().Where(x => x is CodeFunction function && GetOriginalComposedType(function.OriginalLocalMethod.ReturnType) is not null).FirstOrDefault();
+        Assert.True(factoryFunction is not null);
+        writer.Write(factoryFunction);
+        var result = tw.ToString();
+        Assert.Contains("if (parseNode) {", result);
+        Assert.Contains("return parseNode.getNumberValue() || parseNode.getStringValue();", result);
+        Assert.Contains("return undefined;", result);
+        AssertExtensions.CurlyBracesAreClosed(result, 1);
+    }
+
+    [Fact]
+    public async Task Writes_UnionOfObjects_FactoryMethod()
+    {
+        var expectedResultString = @"/**
+ * Creates a new instance of the appropriate class based on discriminator value
+ * @param parseNode The parse node to use to read the discriminator value and create the object
+ * @returns {Cat | Dog}
+ */
+export function createPetsPatchRequestBodyFromDiscriminatorValue(parseNode: ParseNode | undefined) : ((instance?: Parsable) => Record<string, (node: ParseNode) => void>) {
```

Comment:
a full match will be brittle to maintain in case we change something orthogonal like the way we format doc comments, etc...
Can you refactor this test to match parts of the results, like most writer tests are implemented today?